libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -180,6 +180,54 @@ void NavEKF2_core::writeOptFlowMeas(uint8_t &rawFlowQuality, Vector2f &rawFlowRa
     }
 }
 
+// reset variances for mag body state
+void NavEKF2_core::resetMagBodyVariances(void)
+{
+    zeroCols(P,19,21);
+    zeroRows(P,19,21);
+    P[19][19] = sq(frontend->_magNoise);
+    P[20][20] = P[19][19];
+    P[21][21] = P[19][19];
+}
+
+// try changing to another compass
+void NavEKF2_core::tryChangeCompass(void)
+{
+    const auto *compass = _ahrs->get_compass();
+    const uint8_t maxCount = compass->get_count();
```

Comment:
Consider changing maxCount name to reflect it's count of magnetometers: `macCountMagnetomers`